### PR TITLE
Delete users that haven't logged in in over 12 months

### PR DESCRIPTION
**Pull request by emileswarts**
Our data retention policy is that we don't want to keep this data.

### CODE REVIEW

File: test_mysql_connect.py [hunk 1]
```diff
@@ -10,16 +10,45 @@
...
 
-  def test_it_can_connect_to_mysql(self):
-    results = delete_old_users({}, {})
-    assert results is not None
+class TestUserDeletion():
+    @pytest.fixture
+    def setup_users_with_last_logins(self):
+        cursor = db.cursor()
+        cursor.execute('''
+            INSERT INTO userdetails (username, password, email, last_login)
```
Comment (adrianclay) [inline test_mysql_connect.py]: Did you intentionally leave out the ` created_at `?

File: test_mysql_connect.py [hunk 2]
```diff
@@ -10,16 +10,45 @@
...
+        cursor.execute("SELECT COUNT(*) FROM userdetails")
+        user_count = cursor.fetchone()
+        assert user_count[0] == 2
+
+    def test_it_can_delete_users_using_created_at(self, setup_users_without_last_logins):
+        delete_old_users({}, {})
+        cursor = db.cursor()
+        cursor.execute("SELECT COUNT(*) FROM userdetails")
+        user_count = cursor.fetchone()
+        assert user_count[0] == 2
```
Comment (adrianclay) [inline test_mysql_connect.py]: Asserting against the count feels a bit soggy.

Could you assert that selecting the rows gives ` user1 ` and ` user2 ` ?

File: test_mysql_connect.py [hunk 3]
```diff
@@ -10,16 +10,45 @@
...
+            ''')
+        yield
+        cursor.execute("TRUNCATE TABLE userdetails")
+
+    def test_it_can_delete_old_users_using_last_login(self, setup_users_with_last_logins):
+        cursor = db.cursor()
+        delete_old_users({}, {})
+        cursor.execute("SELECT COUNT(*) FROM userdetails")
+        user_count = cursor.fetchone()
+        assert user_count[0] == 2
```
Comment (adrianclay) [inline test_mysql_connect.py]: Similarly with the assert here.

File: user_deletion.py [hunk 4]
```diff
@@ -22,6 +22,12 @@
...
   cursor = db.cursor()
-  cursor.execute("SHOW DATABASES")
-  logger.info("Got some DBs")
-  return cursor.fetchall()
\ No newline at end of file
+
+  cursor.execute('''
+    DELETE FROM userdetails
+      WHERE last_login < DATE_SUB(NOW(), INTERVAL 12 MONTH)
+      OR (last_login IS NULL AND created_at < DATE_SUB(NOW(), INTERVAL 12 MONTH))
```
Comment (adrianclay) [inline user_deletion.py]: Looks good, was this the same query as we performance tested with?

As I recall we were using a limit on the number of rows to keep the query time for each run low.

File: test_mysql_connect.py (hunk 1 above)
Comment (RyanMacG) [reply test_mysql_connect.py]: Yes

File: test_mysql_connect.py (hunk 2 above)
Comment (RyanMacG) [reply test_mysql_connect.py]: It's difficult as the data is returned as a tuple and we'd have to write something custom see if data was included in it. 

I'd counter that we could look for the deleted user and expect the count returned to be 0?

File: user_deletion.py (hunk 4 above)
Comment (RyanMacG) [reply user_deletion.py]: That was for session deletion, we didn't performance test the user table as it was small and well indexed

Comment (RyanMacG) [reply user_deletion.py]: We performance tested extracting the last_login from sessions which was the 3hr+ query

File: test_mysql_connect.py (hunk 2 above)
Comment (adrianclay) [reply test_mysql_connect.py]: Yeah maybe a combination of 2 rows in the DB and the deleted user is missing?  Although 2 asserts feels bad.

Comment (RyanMacG) [reply test_mysql_connect.py]: Would it be possible to move forward on this now and I can loop back on monday to look at the tuple inspection?

File: test_mysql_connect.py [hunk 5]
```diff
@@ -10,16 +11,46 @@
...
+            ''')
+        yield
+        cursor.execute("TRUNCATE TABLE userdetails")
+
+    def test_it_can_delete_old_users_using_last_login(self, setup_users_with_last_logins):
+        cursor = db.cursor()
+        delete_old_users({}, {})
+        cursor.execute("SELECT username FROM userdetails")
+        user_names = [user[0] for user in cursor.fetchall()]
+        assert 'user3' not in user_names
```
Comment (adrianclay) [inline test_mysql_connect.py]: This code doesn't assert that everything wasn't deleted?

**Pull request merged**